File: 1.cpp
#include <iostream>
#include <string>
#include <algorithm>

bool isRightLetters(std::string str) {
    std::string checkword = "TINKOFF";

    if (str.length() != checkword.length()) {
        return false;
    }

    for (char sym : checkword) {
        int index = str.find(sym);
        if (index != std::string::npos) {
            str.erase(index, 1);
        }
        else {
            return false;
        }
    }
    return true;
}

int main() {

    std::string str;
    std::cin >> str;

    isRightLetters(str) ? std::cout << "YES\n" : std::cout << "NO\n";

    return 0;
}

File: 1.js
function isRightLetters (str) {
   
    checkword = 'TINKOFF';

    if(str.length !== checkword.length) {
        return false;
    }

    for(let sym of checkword) {
        const index = str.indexOf(sym);
        if(index !== -1) {
            str = str.slice(0, index) + str.slice(index + 1);
        } else {
            return false;
        }
    }
    return true;
}

const str = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';

isRightLetters(str) ? console.log('YES') : console.log('NO');
isRightLetters('FFOKNIT') ? console.log('YES') : console.log('NO');
isRightLetters('TINKOFFF') ? console.log('YES') : console.log('NO');
isRightLetters('AAAA') ? console.log('YES') : console.log('NO');
isRightLetters('TINKOFF') ? console.log('YES') : console.log('NO');
